core-bundle/src/Security/Authentication/AccessDecisionManager.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Security\Authentication;
+
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AccessDecisionManagerInterface;
+
+class AccessDecisionManager implements AccessDecisionManagerInterface
+{
+    private AccessDecisionManagerInterface $inner;
+    private AccessDecisionManagerInterface $contaoAccessDecisionManager;
+    private ScopeMatcher $scopeMatcher;
+    private RequestStack $requestStack;
+
+    public function __construct(AccessDecisionManagerInterface $inner, AccessDecisionManagerInterface $contaoAccessDecisionManager, ScopeMatcher $scopeMatcher, RequestStack $requestStack)
+    {
+        $this->inner = $inner;
+        $this->contaoAccessDecisionManager = $contaoAccessDecisionManager;
+        $this->scopeMatcher = $scopeMatcher;
+        $this->requestStack = $requestStack;
+    }
+
+    public function decide(TokenInterface $token, array $attributes, $object = null)
+    {
+        $request = $this->requestStack->getCurrentRequest();
```

Comment:
The firewall only operates on the main request (https://github.com/symfony/symfony/blob/a5aaf2c9dc9cc04e472af32d3427a98c5ecebcb8/src/Symfony/Component/Security/Http/Firewall.php#L52) so I think we should check the main request here.

Reply:
Updated accordingly :)